pkg/operator/controller/ingress/dns.go

```diff
@@ -189,3 +193,19 @@ func dnsRecordChanged(current, expected *iov1.DNSRecord) (bool, *iov1.DNSRecord)
 	updated.Spec = expected.Spec
 	return true, updated
 }
+
+// manageDNSForDomain returns true if the given domain matches the baseDomain
+// of the cluster DNS config. It is only used for AWS in the beginning, and will be expanded to other clouds
+// once we know there are no users depending on this.
+// See https://bugzilla.redhat.com/show_bug.cgi?id=2041616
+func manageDNSForDomain(domain string, status *configv1.PlatformStatus, dnsConfig *configv1.DNS) bool {
+	if len(domain) == 0 {
+		return false
+	}
+	switch status.Type {
+	case configv1.AWSPlatformType:
+		return strings.HasSuffix(domain, dnsConfig.Spec.BaseDomain)
```

Comment:
I missed this before, but is `strings.HasSuffix` correct?  I suspect this will consider "foobar.com" to be a subdomain of "bar.com", which isn't correct.

Reply:
Fixed by https://github.com/openshift/cluster-ingress-operator/pull/761/commits/a5d8fe10f800a11ad17974c6527ee960cbe6a52e.